### PR TITLE
Improve development instructions

### PR DESCRIPTION
**Pull request by valeriangalliat**
* Remove potentially dangerous one-liners.
* Describe feature branches and hotfixes.
* Assume the reader is might be forking SassDoc.
* Remove hard reset and force push on `master` when publishing. New feature and hotfixes instructions should prevent this.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -4,10 +4,17 @@
 
 ```sh
 # Clone the repository on your machine
-# If you don't have a SSH key, feel free to clone using HTTPS instead:
-# git clone https://github.com/SassDoc/sassdoc.git
 git clone git@github.com:SassDoc/sassdoc.git
 
+# If you don't have a SSH key, feel free to clone using HTTPS instead
+#git clone https://github.com/SassDoc/sassdoc.git
```
Comment (HugoGiraudel) [inline DEVELOPMENT.md]: Missing space after `#` here. :)

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -4,10 +4,17 @@
...
+
+# If you're not a SassDoc organisation member and you forked the repo,
+# you need to clone `yourname/sassdoc.git` instead.
+#
+# Also add actual SassDoc repo as `upstream` to fetch code updates:
+git remote add upstream git://github.com/SassDoc/sassdoc.git
```
Comment (HugoGiraudel) [inline DEVELOPMENT.md]: GitHub documentation recommends doing it over HTTPS.

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -21,90 +28,159 @@ npm install
...
+# Make sure you have the latest version of SassDoc
 npm update sassdoc -g
 
 # Clone the repository on your machine
-# If you don't have a SSH key, feel free to clone using HTTPS instead:
-# git clone https://github.com/SassDoc/sassdoc-theme-default.git
 git clone git@github.com:SassDoc/sassdoc-theme-default.git
 
+# If you don't have a SSH key, feel free to clone using HTTPS instead
+#git clone https://github.com/SassDoc/sassdoc-theme-default.git
```
Comment (HugoGiraudel) [inline DEVELOPMENT.md]: Missing space after `#` here. :)

**Comment by valeriangalliat** [comment]
Fixed.

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -21,90 +28,159 @@ npm install
...
+```sh
+# Be sure to be on an up-to-date `develop`
+git checkout develop
+git pull
+
+# If you're on a fork, be sure to pull the `upstream` version
+git pull upstream develop
+
+# Create a new feature branch
+git checkout -b feature/my-new-feature
```
Comment (pascalduez) [inline DEVELOPMENT.md]: Is this some kind of convention to prefix a branch with `feature/` ?
Might be a bit confusing regarding remotes and Refs.

Comment (HugoGiraudel) [inline DEVELOPMENT.md]: Yep. Git Flow.

Comment (valeriangalliat) [inline DEVELOPMENT.md]: I'm not really strict about `feature/` and `hotfix/`, it's indeed Git Flow but it's not a problem IMO if this kind of prefix is not used.

Comment (pascalduez) [inline DEVELOPMENT.md]: It's fine then, having clear and established conventions is priceless. :-)

**Comment by pascalduez** [comment]
[LGTM](http://www.lgtm.in/i/MKL5KRvKn)

**Pull request merged**